engine/src/flutter/shell/platform/android/flutter_main.cc

```diff
@@ -324,8 +325,10 @@ AndroidRenderingAPI FlutterMain::SelectedRenderingAPI(
       return kVulkanUnsupportedFallback;
     }
 
-    if (__system_property_find("ro.vendor.mediatek.platform") != nullptr) {
-      // Probably MediaTek. Avoid Vulkan.
+    if (api_level < kMinimumAndroidApiLevelForMediaTekVulkan &&
+        __system_property_find("ro.vendor.mediatek.platform") != nullptr) {
```

Comment:
Should we be using VkPhysicalDeviceProperties to determine this?

Reply:
This is more about the SoC than the GPU driver. early mediatek seems to have equal opportunity bugs.